### PR TITLE
Add reinstall Script / Refactor reset script

### PR DESCRIPTION
**Pull request by robpal94**
## What did you change?

This PR adds a new _reinstall_setup_ - script, which reinstalls your current setup by: 
* Running the existing reset script:
    * Deleting your node_modules
    * Deleting your current `package-lock.json` file
    * Clearing your npm cache
* Reinstalling all required node modules **without creating a new `package-lock.json` file**
* Rebuilding the project

I also added shortcuts to the `package.json` file. You can now simply call 

```bash
npm run reset
```

to reset your current dev - setup or

```bash
npm run reinstall
```

to completely reinstalling your dev setup. 

In addition to that, this PR also refactors the `reset_setup` script a bit. 

## How can others test the changes?
* Run `npm reset` or `npm reinstall`

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: dev_tooling/scripts/reset_setup.bash [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+_VERSION="0.0.2"
```
Comment (cmg-dev) [inline dev_tooling/scripts/reset_setup.bash]: Why is this 0.0.2 and the one above 0.0.1?

Both are new, aren't they?

File: .gitignore [hunk 2]
```diff
@@ -24,11 +24,14 @@ package-lock.json
...
 # Compiled files
-scripts
+scripts/*
 dist
 src/environment.ts
 @fortawesome
 bootstrap
 
+# Allow scripts from dev_tooling
+!dev_tooling/scripts/*
```
Comment (Paulomart) [inline .gitignore]: 👍

File: dev_tooling/scripts/reset_setup.bash [hunk 3]
```diff
@@ -3,13 +3,15 @@
 _VERSION="0.0.1"
 
 ############
-# This Script builds a clean setup of the BPMN - Studio.
-# Keep in mind that your npm cache will be cleaned.
+# This script resets the current setup by deleting the Node Modules,
```
Comment (Paulomart) [inline dev_tooling/scripts/reset_setup.bash]: There is a nice overview in the `reinstall_setup.bash` script, could you move the bullit point list here and mention in the reinstall script that it calls the reset script?

File: dev_tooling/scripts/reset_setup.bash [hunk 4]
```diff
@@ -3,13 +3,15 @@
...
+# the package-lock file and clearing the npm cache.
 ############
 
-# Remove the node modules folder
-echo "Removing Node Modules...."
-rm -rf node_modules/
+# Remove the node modules folder if it exists.
+if [[ -e "node_modules" ]]; then
+  echo "Removing Node Modules...."
```
Comment (Paulomart) [inline dev_tooling/scripts/reset_setup.bash]: Remove one `.`

File: dev_tooling/scripts/reinstall_setup.bash [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+#   * Deleting your node_modules
+#   * Deleting your package-lock file (if it exists)
+#   * Clearing your NPM - Cache
+#   * Reinstalling all required node modules
+#   * Rebuilding
+#
+# Note: Please DO NOT ADD THIS SCRIPT TO THE JENKINSFILE!
+#
+# This script is meant to give you, the human developer, a convenient way, to
+# reinstall your setup. Adding this script to the jenkinsfile would not make
```
Comment (Paulomart) [inline dev_tooling/scripts/reinstall_setup.bash]: `Jenkinsfile`

File: dev_tooling/scripts/reinstall_setup.bash [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+# that much sense, because:
+#   * The jenkins *should* offer a clean setup anyway
```
Comment (Paulomart) [inline dev_tooling/scripts/reinstall_setup.bash]: `Jenkins`

File: dev_tooling/scripts/reinstall_setup.bash [hunk 7]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+_VERSION="0.0.1"
+
+############
+# This script reinstalls the current developement Setup by:
```
Comment (Paulomart) [inline dev_tooling/scripts/reinstall_setup.bash]: `setup`

File: dev_tooling/scripts/reinstall_setup.bash [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+#   * Deleting your node_modules
```
Comment (Paulomart) [inline dev_tooling/scripts/reinstall_setup.bash]: _your_ seems strange here, but I also dont have a better option for you :(

Comment (robpal94) [reply dev_tooling/scripts/reinstall_setup.bash]: Maybe *deletes your node_modules directory* would sound a bit nicer :)

**Comment by cmg-dev** [comment]
@Paulomart bitte approve.

@robpal94 Dann merge ;)

**Pull request merged**